global/install/skm_global_install.sh

```diff
@@ -23,20 +25,75 @@ if [ "$SK_OS" = "macos" ]; then
     LIB_FILE="${SKM_PATH}/lib/macos/libSplashKit.dylib"
     LIB_DEST="/usr/local/lib"
     INC_DEST="/usr/local/include"
+
+    # Check if python3 installed on macOS
+    if command -v python3 &> /dev/null && command -v brew &> /dev/null; then
```

Comment:
This is repeated - lift it outside the platform specifics then you can set it once and just use it later on. The reporting of the error could be within each OS, but the detection outside.

Reply:
Thanks for this prompt! I got caught up in the macOS check needing to be different, but I now realise it can still be better structured to reduce code duplication.